sign/bls/bls.go

```diff
@@ -80,3 +114,15 @@ func hashToPoint(suite pairing.Suite, msg []byte) kyber.Point {
 	x := suite.G1().Scalar().SetBytes(h.Sum(nil))
 	return suite.G1().Point().Mul(x, nil)
 }
+
+func distinct(msgs [][]byte) bool {
+	m := make(map[[32]byte]bool)
+	for _, msg := range msgs {
+		h := sha256.Sum256(msg)
```

Comment:
this takes about 1-2ms for the 100 sigs... I wonder if there's something faster here.

Reply:
a comment explaining why distinctness is required would be helpful here. And a test that confirms that repeated messages are correctly refused.

Reply:
Good ideA

Reply:
I added a comment and a test